packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -1137,16 +1140,6 @@ Directory _getLocalEngineRepo({
   final String abi = _getAbiByLocalEnginePath(engineOutPath);
   final Directory localEngineRepo = fileSystem.systemTempDirectory
     .createTempSync('flutter_tool_local_engine_repo.');
-
-  // Remove the local engine repo before the tool exits.
-  shutdownHooks.addShutdownHook(() {
```

Comment:
IIRC this is not necessary since we delete the shared temp directory on shutdown now.

Reply:
also it wasn't tested so RIP